packages/frames.js/src/lens/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  FrameVerifySignatureResult,
+  LensClient,
+  development,
+} from "@lens-protocol/client";
+
+type LensFrameMessageType = {
+  clientProtocol: string;
+  untrustedData: {
+    specVersion: string;
+    profileId: string;
+    pubId: string;
+    url: string;
+    buttonIndex: number;
+    unixTimestamp: number;
+    deadline: number;
+    inputText: string;
+    state: string;
+    actionResponse: string;
+  };
+  trustedData: {
+    messageBytes: string;
+    identityToken: string;
+  };
+};
+
+export type LensFrameMessageReturnType = LensFrameMessageType & {
+  isActionVerifed: boolean;
+};
+
+export function isLensFrameActionPayload(
+  frameActionPayload: unknown
+): frameActionPayload is LensFrameMessageType {
+  return (
+    typeof frameActionPayload === "object" &&
+    frameActionPayload !== null &&
+    "clientProtocol" in frameActionPayload &&
+    typeof frameActionPayload.clientProtocol === "string" &&
+    frameActionPayload.clientProtocol.startsWith("lens@")
+  );
+}
+
+export async function getLensFrameMessage(
+  frameActionPayload: LensFrameMessageType
+): Promise<LensFrameMessageReturnType> {
+  const lensClient = new LensClient({
+    environment: development,
```

Comment:
should this be hardcoded? should perhaps default to production and can be overridden by a second options parameter in the `getLensFrameMessage` function

Reply:
Good catch, this should default to `production` and I added your suggestion of an `options` parameter (optional) to override this